WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/domainregistration/DomainRegistrationSource.kt

```diff
@@ -74,25 +75,32 @@ class DomainRegistrationSource @Inject constructor(
         siteLocalId: Int,
         selectedSite: SiteModel
     ) {
-        continuation?.cancel()
-        continuation = null
-        coroutineScope.launch(bgDispatcher) {
-            try {
-                val event = suspendCancellableCoroutine<OnPlansFetched> { cancellableContinuation ->
-                    continuation = cancellableContinuation
-                    fetchPlans(selectedSite)
-                }
-                continuation = null
-                if (event.isError) {
-                    val message = "An error occurred while fetching plans : " + event.error.message
-                    appLogWrapper.e(DOMAIN_REGISTRATION, message)
+        if (continuations[siteLocalId] == null) {
+            coroutineScope.launch(bgDispatcher) {
+                try {
+                    val event = suspendCancellableCoroutine<OnPlansFetched> { cancellableContinuation ->
+                        continuations[siteLocalId] = cancellableContinuation
+                        fetchPlans(selectedSite)
+                    }
+                    when {
+                        event.isError -> {
+                            val message = "An error occurred while fetching plans : " + event.error.message
+                            appLogWrapper.e(DOMAIN_REGISTRATION, message)
+                            postState(DomainCreditAvailable(false))
+                        }
+                        siteLocalId == event.site.id -> {
+                            postState(DomainCreditAvailable(isDomainCreditAvailable(event.plans)))
+                        }
+                        else -> {
```

Comment:
Minor (🔍): Next time, please consider adding this extra logic as a part of a separate commit. Else, it can be easily missed.
Suggestion (💡): Consider adding a test for this scenario as well.

Reply:
Good idea. Yeah, until I moved it into the `when` this scenario would have been completed missed. :thu

Reply:
🤔  That scenario is already covered by `when fetched site is different from currently selected site, don't emit value`, but I think I'm going to review all drs tests. I'm not 100% sure they do exactly what they say - even though they all pass as written.

Reply:
Yeap, they do pass, but when a test is passing with or without some main code, then this is a flag that something is not tested. 🤔

Reply:
Agree 100% - I'll be reworking all the tests in a separate PR. I could add them to this PR if you want instead. I don't mind either wwy.